### DISCUSSION TITLE
feature request: Add an option to hide read-only properties from body parameters

### DISCUSSION THREAD

**Discussion by gsmcdonald** [scalar/scalar#1974] [Ideas]
Currently, request body parameters include read-only parameters and show them with a "read-only" tag. It'd be cool to have a configuration option that would allow you to hide the read-only parameters (but only when they are parameters and not when the model is used as output or in the models section).

**Comment by hanspagel** [comment]
Shouldn’t we hide them by default?

**Reply by hanspagel** [reply]
Ah, they are shown in the list, but not in the example request already and you want to hide them in the list, right?

**Example**

https://sandbox.scalar.com/e/MpmhC

**Preview**

![Screenshot 2024-06-07 at 15 24 42](https://github.com/scalar/scalar/assets/1577992/23cc1b96-b2dc-460d-b418-6d623eb23c15)

**Reply by gsmcdonald** [reply]
@hanspagel Yep! Even if they weren't hidden by default (if there were concerns about affecting existing users), just having the option to hide them would be great.

**Reply by hanspagel** [reply]
I think we should add an attribute something, so it’s easy to hide them with CSS:

`<li data-parameter-read-only>…</li>`

@marclave How does that sound?

**Reply by marclave** [reply]
that sounds good :) @hanspagel
